src/data/matrix/basic.lean

```diff
@@ -895,6 +900,8 @@ instance : star_ring (matrix n n α) :=
 
 lemma star_mul (M N : matrix n n α) : star (M ⬝ N) = star N ⬝ star M := star_mul _ _
 
+lemma star_eq_conj_transpose (M : matrix n n α) : star M = Mᴴ := rfl
+
```

Comment:
I think you should add a new `section has_star` above `section star_ring`, add an `instance : has_star (matrix n n α)` (which does not need any of the typeclass assumptions this instance needs), and then put this lemma in that section.

That way you can always use `star` as an alias for `ᴴ` on square matrices.

Reply:
Could you please give the complete code for `instance : has_star (matrix n n α)`. I am not sure if I have correctly understood your suggestion.

Reply:
It would look much like the `instance : star_ring  (matrix n n α)`. If you copy-paste that instance, replace `star_ring` with `has_star`, then clean up the error messages lean give you, then you'll have what I'm suggesting.

Make sure to also run `#lint` after it to check you didn't leave any junk behind.

Reply:
@eric-wieser, why do we only define `instance : has_star (matrix n n α)` for square matrices?
```
class has_star (R : Type u) :=
(star : R → R)
```

Reply:
It seems to me this is either redundant or nor general enough.

Reply:
If you try to define it on non-square matrices you'll find that lean doesn't let you.

Reply:
oh, that is right!